### DISCUSSION TITLE
Feature: add "resolutions" as alias for "overrides"

### DISCUSSION THREAD

**Discussion by csvan** [npm/feedback#499] [CLI]
This may be premature since, to my understanding, overrides are not implemented yet:

https://github.com/npm/rfcs/blob/latest/accepted/0036-overrides.md

However, when they are, it would help teams migrating from Yarn if the `overrides` field could be aliased as `resolutions` as well. Since NPM 7 already supports `yarn.lock` files, this would make for an (almost) seamless transition.

**Comment by ljharb** [comment]
I believe the semantics are quite different, so it would cause lots of breakage to use the same field name.

**Reply by csvan** [reply]
I looked through the RFC, and while `overrides` certainly allow for more usecases than `resolutions`, the places where they overlap seem to function the same way. For example:

```
resolutions: {
    "foo": "1.2.3"
}
```

and

```
overrides: {
    "foo": "1.2.3"
}
```

Would both force a single, top-level install of the "foo" package matching version 1.2.3.

The only think not clear to me is if ranges are supported in overrides, e.g.

```
resolutions: {
    "foo": "^1.2.3"
}
```

and

```
overrides: {
    "foo": "^1.2.3"
}
```

Yarn would interpret the above as installing a single, top-level instance of foo which matches the range ^1.2.3, I do not know if overrides would do the same.

**Reply by ljharb** [reply]
Note that the reverse problem is true, too - there's formats of overrides that would cause yarn to break if it was called "resolutions".

**Reply by csvan** [reply]
Yea that's a good point
